configure/cmake_pch_linux.cmake

```diff
@@ -50,6 +50,7 @@ function(tango_add_pch pch_target ref_target pch_extra_cxx_flags)
         ${pch_compile_opts}
         ${pch_extra_cxx_flags}
         -x c++-header
+        -std=c++14
```

Comment:
@mliszcz That should not be done as that contradicts that we allow the user to set the standard.

Reply:
I couldn't come up with anything better than checking supported compiler flags, value of CXX_STANDARD, and hoping CMake chooses the same flag. We cannot ask CMake what flag it juses for given CXX_STANDARD, see linked issue in commit description. So what we have covers the 'default' case. If user sets different CXX_STANDARD and enables PCH at the same time, the PCH will not be used at all and the user will just get warning like (pch compiled with -std=c++14, source with -std=c++17):
```
main.cpp:1:17: warning: foo.h.gch: not used because `__cpp_static_assert' defined as ` 201411L' not ` 200410L' [-Winvalid-pch]
```
We already got similar problem reported in #734, therefore I opened #733 to remove our hand-made PCH implementation. But this requires upgrading CMake, thus needs further discussions.

The only way I see to fix this is to no use CXX_STANDARD and manually calculate `-std` flag, keep it up to date after each C++ release, handle corner-cases like c++1y, 2a, ..., - basically re-implement what CMake already does.

Maybe you have another idea what to do with those flags?

Reply:
Maybe `-std=c++{CMAKE_CXX_STANDARD}` will be a bit more compatible?

Reply:
> 
> 
> Maybe `-std=c++${CMAKE_CXX_STANDARD}` will be a bit more compatible?


That sounds like a nice compromise while waiting for the move to newer CMake version and the PCH support directly from CMake.

Reply:
> Maybe -std=c++${CMAKE_CXX_STANDARD} will be a bit more compatible?

Yes, let's do that!

Reply:
Done, I modified the commit that introduced this change.

Reply:
@bourtemb what do we do about #733 and CMake's built-in PCH support? If this is something we want to have, I can work on a PR and remove our custom PCH implementation.